modules/apps/fragment/fragment-service/src/main/java/com/liferay/fragment/internal/installer/FragmentFileInstaller.java

```diff
@@ -88,41 +110,19 @@ public int deploy(AutoDeploymentContext autoDeploymentContext)
 			_log.error(exception);
 		}
 		finally {
+			file.delete();
```

Comment:
@yuhai Why the file be deleted? It should not be deleted as it is tracked by file install, next time, if the same file comes, file install will know you need to update it

Reply:
@tinatian, there are reasons to delete the file.

1. Keep the original logic.
2. The main reason is that when deploy fragment, FragmentFileInstaller will execute import action (**add inner file content to the database table**). For example,  In the UI, the end user may edit the data. If not delete the file, after restart the application server(or delete osgi/state folder to restart), FragmentFileInstaller will add related data again and this will override the end user modified data(You may check fragmententryversion table).

I will firstly reopen it in case you missed the message.

cc @dantewang

Thanks,
Hai

Reply:
@yuhai "Try to keep the original logic consistent." What do you mean? You mean we remove the file before your change? where?

So if we did not remove the file before your change, and you think this is wrong, it should be another bug ticket.

What I see here, the before and after your transform is not consistent.

Reply:
@tinatian, please check the first commit from https://github.com/liferay-core-infra/liferay-portal/pull/3494/commits/cab8cf5d596d6b5fdb34e9e87b9eba2639acc874. I removed the below code.
```
for (AutoDeployListener autoDeployListener : _serviceTrackerList) {
			if (autoDeployListener.isDeployable(autoDeploymentContext)) {
				autoDeployListener.deploy(autoDeploymentContext);

				File file = autoDeploymentContext.getFile();

				file.delete();

				return;
			}
		}
```
For the original logic, if the deploy file can be deployed successful by AutoDeployListener, it will delete the file.

Reply:
@dantewang, after restart, for the zip file, it doesn't keep track. module jar should be tracked. If we delete state folder, all files will be tracked again.

Do I need to create one new pull request to separate the commit or push force the change in this pr?

Reply:
@yuhai Yes, you should do it.

@tinatian File Install currently does keep track of files it handled, but it doesn't persist the track, so the track will be cleared after restart.

Reply:
@dantewang, I just push force the changes. No logic changed, so need not to run ci test again.